scripts/dts/gen_defines.py

```diff
@@ -163,6 +163,7 @@ def write_device_extern_header(device_header_out, edt):
         print('extern "C" {', file=dev_header_file)
         print("#endif", file=dev_header_file)
         print("", file=dev_header_file)
+        print("#include <zephyr/device.h>", file=dev_header_file)
```

Comment:
Taking a second look at this PR now that the merge window is open. Why is this needed?

You should never need to include device_extern.h directly; it's just an internal sub-header of device.h.